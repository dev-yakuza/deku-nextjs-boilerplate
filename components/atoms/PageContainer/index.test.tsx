import { render, screen } from '@testing-library/react'

import { PageContainer } from '.'

describe('<PageContainer />', () => {
  it('Rendered well with string', async () => {
    const { container } = render(
      <PageContainer>This is PageContainer</PageContainer>,
    )

    const pageContainer = screen.getByText('This is PageContainer')
    expect(pageContainer).toHaveStyle(`
      height: 100vh;
      display: flex;
      flex-direction: column;
      background-color: #eee;
      overflow: scroll;
    `)

    expect(container).toMatchSnapshot()
  })

  it('Rendered well with component', async () => {
    const { container } = render(
      <PageContainer>
        <h1>This is PageContainer</h1>
      </PageContainer>,
    )

    const pageContainer = screen.getByText('This is PageContainer')
    expect(pageContainer.tagName).toBe('H1')

    expect(container).toMatchSnapshot()
  })
})
