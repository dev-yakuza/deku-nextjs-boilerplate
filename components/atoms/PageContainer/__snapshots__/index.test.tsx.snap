// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageContainer /> Rendered well with component 1`] = `
.emotion-0 {
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #eee;
  overflow: scroll;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <h1>
      This is PageContainer
    </h1>
  </div>
</div>
`;

exports[`<PageContainer /> Rendered well with string 1`] = `
.emotion-0 {
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #eee;
  overflow: scroll;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    This is PageContainer
  </div>
</div>
`;
